Integration not compatible with Autofac 5.0 Release Candidate

The Autofac.Mvc integration has build errors when built against the latest Autofac Release candidate in MyGet, specifically:

```
src\Autofac.Integration.Mvc\RegistrationExtensions.cs(457,44,457,60): error CS1501: No overload for method 'ResolveComponent' takes 2 arguments
```

The IComponentContext.ResolveComponent method has changed signature, which breaks one specific bit of code.

In addition, I believe the NuSpec will need updating to change the version constraints from ``[3.5.0,5.0.0)`` to ``[5.0.0, 6.0.0)``, because this version will no longer be compatible with older Autofac versions.

--- a/NuGet.Config
+++ b/NuGet.Config
@@ -1,9 +1,9 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <clear/>
-    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
-    <add key="Xunit MyGet" value="https://myget.org/F/xunit/api/v3/index.json" />
+    <clear />
+    <add key="Autofac MyGet" value="https://www.myget.org/F/autofac/api/v2" />
+    <add key="NuGet v3" value="https://api.nuget.org/v3/index.json" />
   </packageSources>
   <disabledPackageSources>
     <add key="Microsoft and .NET" value="true" />

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,6 +1,6 @@
 environment:
-  package_semantic_version: 4.0.2
-  assembly_semantic_version: 4.0.0
+  package_semantic_version: 5.0.0
+  assembly_semantic_version: 5.0.0
 
 version: $(package_semantic_version).{build}
 

--- a/src/Autofac.Integration.Mvc/Autofac.Integration.Mvc.csproj
+++ b/src/Autofac.Integration.Mvc/Autofac.Integration.Mvc.csproj
@@ -48,6 +48,7 @@
   <ItemGroup>
     <Reference Include="Autofac, Version=5.0.0.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Autofac.5.0.0-develop-00611\lib\net45\Autofac.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Web.Infrastructure, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <Private>True</Private>

--- a/src/Autofac.Integration.Mvc/Autofac.Integration.Mvc.csproj
+++ b/src/Autofac.Integration.Mvc/Autofac.Integration.Mvc.csproj
@@ -46,9 +46,8 @@
     <AssemblyOriginatorKeyFile>..\..\Autofac.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Autofac, Version=4.0.1.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Autofac.4.0.1\lib\net45\Autofac.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Autofac, Version=5.0.0.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\Autofac.5.0.0-develop-00611\lib\net45\Autofac.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Web.Infrastructure, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <Private>True</Private>

--- a/src/Autofac.Integration.Mvc/Autofac.Integration.Mvc.nuspec
+++ b/src/Autofac.Integration.Mvc/Autofac.Integration.Mvc.nuspec
@@ -14,7 +14,7 @@
     <iconUrl>https://cloud.githubusercontent.com/assets/1156571/13684110/16b8f152-e6bf-11e5-84ae-22c66c6d351a.png</iconUrl>
     <releaseNotes>Release notes are at https://github.com/autofac/Autofac.Mvc/releases</releaseNotes>
     <dependencies>
-      <dependency id="Autofac" version="[4.0.1,5.0.0)" />
+      <dependency id="Autofac" version="[5.0.0,6.0.0)" />
       <dependency id="Microsoft.AspNet.Mvc" version="[5.1.0,6.0.0)" />
     </dependencies>
   </metadata>

--- a/src/Autofac.Integration.Mvc/RegistrationExtensions.cs
+++ b/src/Autofac.Integration.Mvc/RegistrationExtensions.cs
@@ -444,27 +444,35 @@ namespace Autofac.Integration.Mvc
 
             return registration
                 .ExternallyOwned()
-                .OnRegistered(e => e.ComponentRegistry.Register(RegistrationBuilder
-                    .ForDelegate((c, p) =>
-                    {
-                        var session = HttpContext.Current.Session;
-                        object result;
-                        lock (session.SyncRoot)
+                .OnRegistered(e =>
+                {
+                  foreach (var service in services)
+                  {
+                    e.ComponentRegistry.Register(RegistrationBuilder
+                        .ForDelegate((c, p) =>
                         {
+                          var session = HttpContext.Current.Session;
+                          object result;
+                          lock (session.SyncRoot)
+                          {
                             result = session[e.ComponentRegistration.Id.ToString()];
                             if (result == null)
                             {
-                                result = c.ResolveComponent(e.ComponentRegistration, p);
-                                session[e.ComponentRegistration.Id.ToString()] = result;
+                              var resolveRequest = new ResolveRequest(service, e.ComponentRegistration, p);
+                              result = c.ResolveComponent(resolveRequest);
+                              session[e.ComponentRegistration.Id.ToString()] = result;
                             }
-                        }
-                        return result;
-                    })
-                    .As(services)
-                    .InstancePerLifetimeScope()
-                    .ExternallyOwned()
-                    .CreateRegistration()));
-        }
+                          }
+
+                          return result;
+                        })
+                        .As(service)
+                        .InstancePerLifetimeScope()
+                        .ExternallyOwned()
+                        .CreateRegistration());
+                  }
+                });
+    }
 
         /// <summary>
         /// Inject an IActionInvoker into the controller's ActionInvoker property.

--- a/src/Autofac.Integration.Mvc/packages.config
+++ b/src/Autofac.Integration.Mvc/packages.config
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Autofac" version="4.0.1" targetFramework="net45" />
+  <package id="Autofac" version="5.0.0-develop-00611" targetFramework="net45" />
   <package id="Microsoft.AspNet.Mvc" version="5.1.0" targetFramework="net45" />
   <package id="Microsoft.AspNet.Razor" version="3.1.0" targetFramework="net45" />
   <package id="Microsoft.AspNet.WebPages" version="3.1.0" targetFramework="net45" />

--- a/test/Autofac.Integration.Mvc.Test/Autofac.Integration.Mvc.Test.csproj
+++ b/test/Autofac.Integration.Mvc.Test/Autofac.Integration.Mvc.Test.csproj
@@ -47,6 +47,7 @@
   <ItemGroup>
     <Reference Include="Autofac, Version=5.0.0.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Autofac.5.0.0-develop-00611\lib\net45\Autofac.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="Castle.Core, Version=3.3.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Castle.Core.3.3.3\lib\net45\Castle.Core.dll</HintPath>

--- a/test/Autofac.Integration.Mvc.Test/Autofac.Integration.Mvc.Test.csproj
+++ b/test/Autofac.Integration.Mvc.Test/Autofac.Integration.Mvc.Test.csproj
@@ -45,9 +45,8 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Autofac, Version=4.0.1.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Autofac.4.0.1\lib\net45\Autofac.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Autofac, Version=5.0.0.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\Autofac.5.0.0-develop-00611\lib\net45\Autofac.dll</HintPath>
     </Reference>
     <Reference Include="Castle.Core, Version=3.3.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Castle.Core.3.3.3\lib\net45\Castle.Core.dll</HintPath>

--- a/test/Autofac.Integration.Mvc.Test/packages.config
+++ b/test/Autofac.Integration.Mvc.Test/packages.config
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Autofac" version="4.0.1" targetFramework="net45" />
+  <package id="Autofac" version="5.0.0-develop-00611" targetFramework="net45" />
   <package id="Castle.Core" version="3.3.3" targetFramework="net451" />
   <package id="Microsoft.AspNet.Mvc" version="5.1.0" targetFramework="net45" />
   <package id="Microsoft.AspNet.Razor" version="3.1.0" targetFramework="net45" />

